doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,237 @@
+\ingroup module_hidden
+\page jbmc-user-manual JBMC User Manual
+
+
+\section jbmc-manual-intro Introduction
+
+JBMC is a bounded model checking tool for the Java language. It follows the
+same principles as CBMC, thus the existing
+[documentation for CBMC](http://www.cprover.org/cprover-manual/cbmc.html)
+largely applies to JBMC.
+
+In brief, JBMC creates a model from Java Bytecode (given as input as a .class
+file or a .jar) and performs static analysis and symbolic simulation which allow
+to prove or refute properties. These properties are either automatically
+generated, or provided by the user in the form of assertions.
+
+Using examples, the following section will guide you through the basic
+features of JBMC, and explore the most useful features.
+
+
+\section jbmc-tutorial JBMC tutorial
+
+\subsection jbmc-manual-auto-assert Automatic assertions
+
+Consider the following simplistic `ExampleArray.java` file which we have compiled
+into `ExampleArray.class` and which contains a utility method to query a given
+array at a given index:
+```
+ 1| package tutorial;
+ 2|
+ 3| public class ExampleArray {
+ 4|
+ 5|     public static int query(int[] array, int index) {
```

Comment:
It might be worth changing this to static void main(String[]) to show also the case without --function option

Reply:
I think it's slightly confusing to start with the example where the entry point is main, because the reader needs to understand that the entry point is always a function, and that `main` is only the default. So I change one of the later examples (the one with models). Furthermore, that example is better suited to be a main because we don't use any arguments.

Reply:
It is very intuitive that the entry point is `main` because this is how you run a Java program. You can then then explain that you can choose other methods entry point (which is unlike `java`).

Reply:
That's right I'll try to start with an example with `main` as entry point then.